integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -3900,6 +3902,265 @@ mod access_key_nonce_range_tests {
         assert_eq!(num_requests, 4 * 20);
     }
 
+    struct ChunkForwardingOptimizationTestData {
+        num_validators: usize,
+        env: TestEnv,
+
+        num_part_ords_requested: usize,
+        num_part_ords_sent_as_partial_encoded_chunk: usize,
+        num_part_ords_forwarded: usize,
+        num_forwards_with_missing_chunk_header: usize,
+        chunk_parts_that_must_be_known: HashSet<(ChunkHash, u64, usize)>,
+    }
+
+    impl ChunkForwardingOptimizationTestData {
+        fn new() -> ChunkForwardingOptimizationTestData {
+            let num_clients = 4;
+            let num_validators = 4 as usize;
+            let num_block_producers = 1;
+            let epoch_length = 10;
+
+            let accounts: Vec<AccountId> =
+                (0..num_clients).map(|i| format!("test{}", i).parse().unwrap()).collect();
+            let mut genesis = Genesis::test(accounts, num_validators as u64);
+            {
+                let config = &mut genesis.config;
+                config.epoch_length = epoch_length;
+                config.shard_layout = ShardLayout::v1_test();
+                config.num_block_producer_seats_per_shard = vec![
```

Comment:
I think you don't need this field now that chunk only producers are enabled

Reply:
There's an assertion that this vector has the same length as the number of shards.